### ISSUE TITLE
resolve PROTOTYPE comments regarding target-typed conditional expression

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#46171]
The following reflect previous action items recorded in the C# compiler for target-typed conditional expressions. We need to determine what, if any, actions are required to address them.

-----
in `CSharpSemanticModel.cs`
```csharp
                else if (highestBoundExpr is BoundConditionalOperator { WasTargetTyped: true } cond)
                {
                    type = cond.NaturalTypeOpt;
                    convertedType = cond.Type;
                    convertedNullability = nullability;
                    // PROTOTYPE(ngafter): Should we preserve the set of conversions?
                    // By now they have been pushed into the subexpressions.
                    conversion = Conversion.MakeConditionalExpression(ImmutableArray<Conversion>.Empty);
                }
```
-----
in `IToperationTests_IThrowOperation.cs`
```csharp
            // PROTOTYPE(ngafter): Investigate the tree in block B4 (syntax for the operand of the conversion)
...
    Block[B4] - Block [UnReachable]
        Predecessors (0)
        Statements (1)
            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'x = y ? thr ...  throw ex2;')
              Expression: 
                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Boolean, IsInvalid) (Syntax: 'x = y ? thr ... : throw ex2')
                  Left: 
                    IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Boolean, IsImplicit) (Syntax: 'x')
                  Right: 
                    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Boolean, IsInvalid, IsImplicit) (Syntax: 'throw ex2')
                      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                        (ImplicitThrow)
                      Operand: 
                        IOperation:  (OperationKind.None, Type: null, IsInvalid, IsImplicit) (Syntax: 'throw ex2')
```
-----
in `IToperationTests_IThrowOperation.cs`
```csharp
            // PROTOTYPE(ngafter): Investigate the tree in block B4.  The conversion should be an identity conversion on the whole conditional.
...
    Block[B4] - Block [UnReachable]
        Predecessors (0)
        Statements (1)
            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'x = y ? thr ...  throw ex2;')
              Expression: 
                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Boolean) (Syntax: 'x = y ? thr ... : throw ex2')
                  Left: 
                    IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Boolean, IsImplicit) (Syntax: 'x')
                  Right: 
                    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Boolean, IsImplicit) (Syntax: 'throw ex2')
                      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                        (ImplicitThrow)
                      Operand: 
                        IOperation:  (OperationKind.None, Type: null, IsImplicit) (Syntax: 'throw ex2')
```

**Comment by RikkiGibson** [comment]
@jaredpar do you want this addressed in 16.8 or in a future release.

**Comment by jaredpar** [comment]
@RikkiGibson this slipped my radar, we'll fix in 16.9
